kotlin/edit-features-with-feature-linked-annotation/src/main/java/com/esri/arcgisruntime/sample/editfeatureswithfeaturelinkedannotation/MainActivity.kt

```diff
@@ -0,0 +1,250 @@
+        movePolylineVertex(mapPoint)
+      } else {
+        movePoint(mapPoint)
+      }
+    } else {
+      selectFeature(screenPoint)
+    }
+  }
+
+  /**
+   * Select a feature near the given screen point using identify and, for a point feature, show a
+   * dialog to edit attributes. Future taps will call move functions.
+   *
+   * @param screenPoint at which to select a feature
+   */
+  private fun selectFeature(screenPoint: android.graphics.Point) {
+    // clear any previously selected features
+    clearSelection()
+    selectedFeature = null
+
+    // identify across all layers
+    val identifyLayerResultFuture = mapView.identifyLayersAsync(screenPoint, 10.0, false)
+    identifyLayerResultFuture.addDoneListener {
+      // get the list of result from the future
+      val identifyLayerResult = identifyLayerResultFuture.get()
+      // for each layer from which an element was identified
+      identifyLayerResult.forEach { result ->
+        // check if the layer is a feature layer, thereby excluding annotation layers
+        (result.layerContent as? FeatureLayer)?.let { featureLayer ->
+          // get a reference to the identified feature
+          selectedFeature = result.elements[0] as? Feature
+          selectedFeature?.also {
+            // select the feature
+            featureLayer.selectFeature(it)
+            when (it.geometry.geometryType) {
+              // when selected feature is a point, show editable attributes
+              GeometryType.POINT -> showEditableAttributes(it)
+              // when selected feature is a polyline,
+              GeometryType.POLYLINE -> selectedFeatureIsPolyline = true
+              else -> Log.e(TAG, "Feature of unexpected geometry type selected")
+            }
+            // return, since a feature has been selected
+            return@addDoneListener
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and
+   * 'ST_STR_NAM' attributes.
+   *
+   * @param selectedFeature whose attributes will be edited
+   */
+  private fun showEditableAttributes(selectedFeature: Feature) {
+    // inflate the edit attribute layout
+    val editAttributeView = layoutInflater.inflate(R.layout.edit_attribute_layout, null)
+    // create an alert dialog
+    AlertDialog.Builder(this).apply {
+      setTitle("Edit feature attribute:")
+      setView(editAttributeView)
+      // populate edit texts with current attribute values
+      editAttributeView.addressNumberEditText.setText(selectedFeature.attributes["AD_ADDRESS"].toString())
+      editAttributeView.streetEditText.setText(selectedFeature.attributes["ST_STR_NAM"].toString())
+      setPositiveButton("OK") { _, _ ->
+        // set AD_ADDRESS value to the int from the edit text
+        selectedFeature.attributes["AD_ADDRESS"] =
+          editAttributeView.addressNumberEditText.text.toString().toInt()
```

Comment:
I think you're 100% right but I just can't believe this is how you get an int from an edittext